lua/arrow/buffer_persist.lua

```diff
@@ -61,8 +81,11 @@ function M.sync_buffer_bookmarks(bufnr)
 	local file = io.open(path, "w")
 
 	if file then
-		file:write(json.encode(M.local_bookmarks[bufnr]))
-
+		if M.local_bookmarks[bufnr] == nil or #M.local_bookmarks[bufnr] == 0 then
```

Comment:
Sometims it will writes "null" to json file that broke the bookmarks loading, instead to always writing [].